### DISCUSSION TITLE
Team / Group permissions, invite, ...

### DISCUSSION THREAD

**Discussion by JPustkuchen** [directus/directus#22017] [Feature Requests]
### Summary

Provide a Group / Team functionality to handle permissions.

In larger SaaS applications you typically need to allow access to other users by inviting them into the project. These members may already exist or not.
Groups can help to generalize and structure these permission assignments, for example if a group represents an external company. In real world you could add a group "Our helpful marketing agency" to the project and allow the team lead of the marketing company ("Group Manager") to manage the roles within this group. For your project this group could have a permission limit that can be assigned.
In another example, a standard group could be used to add other members of your team and assign their basic roles.

It would be a game-changer to have a concept for this in directus!

Appwrite has a basic functionality for this, but maybe it can be done even better and more flexible:
https://appwrite.io/docs/products/auth/teams

### Basic Example

Take a look at similar functionality in GitHub, GitLab, Google Ads, ...
The Drupal [Group](https://www.drupal.org/project/group) Module might also be a good example for a flexible implementation of such a functionality.

### Motivation

A group feature is a complex functionality, but essential in most larger SaaS applications. It would put the User / Roles / Permission concept on another level and for directus it could be a further nice USP!

### Detailed Design

TBD

### Must Have:
- Create groups (if the user has permission to do so (within a certain scope))
- Add / remove users in groups
- Set roles for group members

### Should Have:
- Role settings
- Create custom roles within the group and assign them permissions
- Assign groups to other entities (e.g. projects)
- Invite users into groups
- Set default permissions for new group members

### Could Have:
- Default group roles (like Group manager, Member, Guest)
- Start / end date for group membership
- Apply for role membership (via secret link?)

### Drawbacks

This should be optional, so it can be added, if the complexity is needed.

### Alternatives

TBD

### Adoption Strategy

TBD

**Comment by oceanvievv** [comment]
+1

**Comment by g-vellis-se** [comment]
I am surprised that the given functionality is not already supported.
